### ISSUE TITLE
Add a license

### ISSUE THREAD

**Issue by tkw1536** [KWARC/MMT#1]
We should add a license.

**Comment by cmaeder** [comment]
I've looked at http://www.gnu.org/licenses/gpl-howto.html and saw "the process involves adding two elements to each source file of your program: a copyright notice (such as “Copyright 1999 Terry Jones”), and a statement of copying permission".

I find the "statement of copying permission" too long. Most of our files do not even have the copyright notice but maybe it can be generated from the repository commits. (But then we have not only .scala source files and some (copied) source files may have their own license )

**Comment by kohlhase** [comment]
we really should have a boilerplate header in all files in MMT. You do not really have to have all that text it is fully sufficient to have something like 

```
This file is part of the MMT API (see http://...)
Copyright (c) 2015 by ????? 
Licensed under GPL3, see http://www.gnu.org/licenses/gpl.html
```
that is not too bad.

**Comment by kohlhase** [comment]
One thing that can be added to these three lines is a statement where the original file is, but this would be optional

**Comment by cmaeder** [comment]
I agree that there should be a header in every file. However, I would prefer a more liberal license without "copyleft" like BSD3, BSD2 or the MIT License https://en.wikipedia.org/wiki/BSD_licenses.

Do we have a default copyright holder (like KWARC group, Florian Rabe or Jacobs University)?

**Comment by kohlhase** [comment]
As the MMT API is almost exclusively going to be used in academic settings, but GPL3 is the standard choice. Having a copyleft license gives us the chance to use weaker licenses for "clients" who have problems with copyleft and possibly use that as a negotiating chip. 

> Do we have a default copyright holder (like KWARC group, Florian Rabe or Jacobs University)?

that is the real question around here. Whoever has the copyright can re-license, but in this situation this is for Florian to decide. 

I do not think that Jacobs University should have the copyright (lest they get ideas about commercialization). For all the other probjects we have always used "The KWARC Group http://kwarc.info" as a group copyright.

**Comment by cmaeder** [comment]
I've added the GPL3 text. https://svn.kwarc.info/repos/MMT/COPYING

Several names (or groups) could follow after "Copyright".

**Comment by florian-rabe** [comment]
I've added one, which needs discussion now.
